package.json

```
@@ -45,6 +45,14 @@
     "workspaceContains:**/scala/**"
   ],
   "contributes": {
+    "rootPatterns": [
+      {
+        "filetype": "scala",
+        "patterns": [
+          "build.sbt"
```

Instead of just including `build.sbt` can we maybe include the build files for the other supported tools as well, and then still end with `.git`?

```suggestion
          "build.sbt", "build.sc", "build.gradle", "pom.xml", ".git"
```

---

I think `.git` should be in `coc.preferences.rootPatterns`

---

ie default rootpatterns

---

Yea, I know it defaults to `[".git", ".hg", ".projections.json"]`, but I wasn't sure if when a user sets those values if they are set on top of the defaults, or they fully replace everything in the `rootPatterns` 🤔

---

Yea it does look like the defaults are added to rather than replaces. I'd say remove the `.git`, but add in the others 👍

---

yup, the rootpatterns logic is a bit complicated https://github.com/neoclide/coc.nvim/blob/master/src/workspace.ts